lib/ruby_lsp/ruby_lsp_rails/runner_client.rb

```diff
@@ -149,6 +149,8 @@ def shutdown
         send_message("shutdown")
         sleep(0.5) # give the server a bit of time to shutdown
         [@stdin, @stdout, @stderr].each(&:close)
+      rescue IOError
+        # The server connection may have died
```

Comment:
I wonder if we should still issue a kill signal to guarantee we don't leave orphan processes. Could there be a scenario where we get an `IOError`, but the child process is still alive?

Maybe we can do this inside the rescue?
```ruby
Process.kill(T.must(Signal.list["KILL"]), @wait_thread.pid)
```

We can also ask the reporter to modify their local installation of `ruby-lsp-rails` so that we can verify that it works.